src/Analyzers/CSharp/Analyzers/NewLines/ArrowExpressionClausePlacement/ArrowExpressionClausePlacementDiagnosticAnalyzer.cs

```diff
@@ -36,7 +36,10 @@ private void AnalyzeTree(SyntaxTreeAnalysisContext context)
             if (option.Value)
                 return;
 
-            Recurse(context, option.Notification.Severity, context.Tree.GetRoot(context.CancellationToken));
+            var root = context.Tree.GetRoot(context.CancellationToken);
+            if (context.FilterSpan.HasValue)
+                root = root.FindNode(context.FilterSpan.GetValueOrDefault());
+            Recurse(context, option.Notification.Severity, root);
```

Comment:
so i don't hate this... but i also don't love it either... trying to figure out why.  I think it's because it really presumes that the node you get from intersecting the filter span necessarily will at least have the member that this analyzer is for...

Maybe that's ok though.  I have a nagging feeling in the back of my head that this may end up choosing a wrong node that ends up not working out.  But i'm fine going ahead and revising if we run into any issues.